src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -180,26 +180,54 @@ protected void WriteFullObjectJson(JsonWriter writer, object value, JsonSerializ
                         writer.WritePropertyName(prop.PropertyName);
                         serializer.Serialize(writer, id);
                         break;
+
                     case PropertyNames.Links:
                     case PropertyNames.Meta:
                         writer.WritePropertyName(prop.PropertyName);
                         serializer.Serialize(writer, propValue);
                         break;
+
                     case PropertyNames.Type:
                         writer.WritePropertyName("type");
                         type = typeProp?.ValueProvider?.GetValue(value) ?? GenerateDefaultTypeName(valueType);
                         serializer.Serialize(writer, type);
                         break;
+
                     default:
-                        //we do not know if it is an Attribute or a Relationship
-                        //so we will send out a probe to determine which one it is
-                        var probe = new AttributeOrRelationshipProbe(writer);
-                        probe.WritePropertyName(prop.PropertyName);
-                        WriterUtil.WritePropertyValue(serializer, prop, propValue, probe);
-
-                        (probe.PropertyType == AttributeOrRelationshipProbe.Type.Attribute
-                            ? attributes
-                            : relationships).Add(probe);
+                        switch (propValue)
+                        {
+                            case string _:
+                                var k = prop;
+                                attributes.Add(x => x.WritePropertyName(k.PropertyName));
+                                attributes.Add(x => x.WriteValue(propValue));
+
+                                //writer.WritePropertyName(prop.PropertyName);
+                                //serializer.Serialize(writer, propValue);

```

Comment:
Generally I try to push as much as I can through the `serializer.Serialiaze` as it ensures it goes through json.net, applying any specific setting, for example a JsonConvertor for `string`

For strings this would probably not be an issue and would be happy to give up that functionality if we got more useful funcitonality in return, however this would only seem to produce a very mild performance improvement